service/src/main/java/org/cbioportal/service/StudyViewService.java

```diff
@@ -1,9 +1,26 @@
 package org.cbioportal.service;
 
+import org.cbioportal.model.AlterationCountByGene;
+import org.cbioportal.model.CopyNumberCountByGene;
 import org.cbioportal.model.GenomicDataCount;
+import org.cbioportal.service.exception.StudyNotFoundException;
+import org.springframework.cache.annotation.Cacheable;
 
 import java.util.List;
 
 public interface StudyViewService {
+    @Cacheable(cacheResolver = "generalRepositoryCacheResolver", condition = "@cacheEnabledConfig.getEnabled()")
```

Comment:
@kalletlak i'm confused because this looks like general repository not static (issue title says static)

Reply:
it should be general cache. I'll rename it